osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Framework.Localisation;
+using osu.Game.Beatmaps;
+using osu.Game.Collections;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Overlays;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Screens.Select.FooterV2
+{
+    public partial class BeatmapOptionsPopover : OsuPopover
+    {
+        private FillFlowContainer buttonFlow = null!;
+        private readonly FooterButtonOptionsV2 footerButton;
+
+        private WorkingBeatmap beatmapWhenOpening = null!;
+
+        public BeatmapOptionsPopover(FooterButtonOptionsV2 footerButton)
+        {
+            this.footerButton = footerButton;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ManageCollectionsDialog? manageCollectionsDialog, SongSelect? songSelect, OsuColour colours, IBindable<WorkingBeatmap> beatmap)
+        {
+            Content.Padding = new MarginPadding(5);
+
+            Child = buttonFlow = new FillFlowContainer
+            {
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Vertical,
+                Spacing = new Vector2(3),
+            };
+
+            beatmapWhenOpening = beatmap.Value;
+
+            addHeader("General");
+            addButton(@"Manage collections", FontAwesome.Solid.Book, () => manageCollectionsDialog?.Show());
+
+            addHeader("For all difficulties", beatmapWhenOpening.BeatmapSetInfo.ToString());
+            addButton(@"Delete beatmap", FontAwesome.Solid.Trash, () => songSelect?.DeleteBeatmap(), colours.Red1);
+
+            addHeader("For selected difficulty", beatmapWhenOpening.BeatmapInfo.DifficultyName);
+            // TODO: make work, and make show "unplayed" or "played" based on status.
+            addButton(@"Mark as played", FontAwesome.Regular.TimesCircle, null);
+            addButton(@"Hide", FontAwesome.Solid.Magic, null);
```

Comment:
This hide button addition makes edit beatmap now index 6. It could be commented, but it's already possible to hide.

Reply:
Have moved it to the end.

Reply:
Oh yeah, this would break my muscle memory of `F3` `5` for edit.